cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -3842,6 +3842,7 @@ mod tests {
                 "-9223372036854775809",
                 ToASTErrorKind::IntegerLiteralTooLarge(9223372036854775809),
             ),
+            // This test doesn't fail with an internal representation of i128:
             // Contrary to Rust, this expression is not valid because the
             // parser treats it as a negation operation whereas the operand
             // (9223372036854775808) is too large.
```

Comment:
can we reverse the order here -- first explain why the test has the behavior it does by default (which is the "Contrary to Rust" part) -- then we can have an additional comment about the behavior with i128 (and actually, why doesn't it fail with i128?)